crates/bevy_reflect/derive/src/lib.rs

```diff
@@ -575,7 +575,8 @@ pub fn impl_reflect_value(input: TokenStream) -> TokenStream {
 }
 
 /// A replacement for `#[derive(Reflect)]` to be used with foreign types which
-/// the definitions of cannot be altered.
+/// the definitions of cannot be altered. It is used internally by the `bevy_reflect` crate and
+/// cannot be used by other crates due to the orphan rule.
```

Comment:
The tricky thing here is that it _can_ be used by other crates, just not for foreign types (i.e. types defined outside your own crate).

For example, we use it on `glam` types such that we can control the serialized representation of the struct (e.g. as `(x: 1., y: 2., z: 3.,)` rather than `(1., 2., 3.,)`). If `glam` decided they wanted to add a dedicated `bevy` feature and do this themselves, they'd likely also want use `impl_reflect!` to achieve the same effect.

So perhaps we should say something more to the effect of "cannot be used to break Rust's orphan rule"? Or maybe even rewording that first sentence to either avoid "foreign" or mentioning that only `bevy_reflect` is capable of using this on foreign types?

Reply:
To add to this, it probably still is better to suggest that this macro is more so internal to `bevy_reflect` like you already have. Again, you can use it in your own crate, but it's almost always preferable to just use `#[derive(Reflect)]`.